python_modules/dagster/dagster/_daemon/auto_run_reexecution/auto_run_reexecution.py

```diff
@@ -152,8 +139,27 @@ def retry_run(
         )
     )
 
-    strategy = get_reexecution_strategy(failed_run, instance) or DEFAULT_REEXECUTION_POLICY
+    _, run_group = check.not_none(instance.get_run_group(failed_run.run_id))
+    run_group_list = list(run_group)
+
+    # it is possible for the daemon to die between creating the run and submitting it. We account for this
+    # possibility by checking if the a run already exists in the run group with the parent run id of the
+    # failed run and resubmit it if necessary.
+    existing_retried_run = get_retried_run_if_exists(failed_run, run_group_list)
```

Comment:
should we move the AUTO_RETRY_RUN_ID_TAG tag check into this method and make get_automatically_retried_run_if_exists? It seems like the current impl would also return a manually retried run and short-circuit in the (admittedly unlikely) event that one was created before this auto-retry run kicked off, but i think we can pretty much just ignore manually retried runs for the purpose of this "check if the daemon died" check right?

Like i'm imagining:
- for whatever reason the auto retry daemon is delayed a few minutes
- user gets impatient, launches a manually retried run
- this check will now detect the manually retried run and not automatically retry right? Or if the manually retried run happens to be in a NOT_STARTED state, it will submit it again (even though it might be already being submitted by some other process)

Reply:
> for whatever reason the auto retry daemon is delayed a few minutes
user gets impatient, launches a manually retried run
this check will now detect the manually retried run and not automatically retry right? Or if the manually retried run happens to be in a NOT_STARTED state, it will submit it again (even though it might be already being submitted by some other process)


yeah that is the behavior that will happen both in master and in this PR as it stands right now 

If we change `get_retried_run_if_exists` to `get_automatically_retried_run_if_exists` are you thinking the impl would be like
```python
def get_automaticall_retried_run_if_exists(
    run: DagsterRun, instance,
) -> Optional[DagsterRun]:
    if run.tags.get(AUTO_RETRY_RUN_ID_TAG, None) is not None:
         return instance.get_run_by_id(run.tags[AUTO_RETRY_RUN_ID]
```

because then i think we still have to deal with the edge case where the daemon fails between creating the run and adding the tag to the original run

so maybe 
```python
def get_automaticall_retried_run_if_exists(
    run: DagsterRun, run_group: Sequence[DagsterRun], instance
) -> Optional[DagsterRun]:
    if run.tags.get(AUTO_RETRY_RUN_ID_TAG, None) is not None:
         return instance.get_run_by_id(run.tags[AUTO_RETRY_RUN_ID]
   child_run = next(
        (retried_run for retried_run in run_group if run.run_id == retried_run.parent_run_id), None
    )
   if child_run.status == DagsterRunStatus.NOT_STARTED:
      # if the run is in NOT_STARTED status then assume the daemon failed between creating the run and adding the tag to the parent run
      instance.add_run_tags(run.run_id, {AUTO_RETRY_RUN_ID_TAG: child_run.run_id}
     return child run
```

Then there's still a chance that the manual retry is timed just right to be NOT_STARTED when this fn runs, but that seems so unlikely